reactor-netty-http/src/main/java/reactor/netty/http/client/Http2Pool.java

```diff
@@ -635,12 +603,8 @@ Slot findConnection(ConcurrentLinkedQueue<Slot> resources) {
 		return null;
 	}
 
-	boolean maxIdleReached(Slot slot) {
-		return maxIdleTime != -1 && slot.idleTime() >= maxIdleTime;
-	}
-
-	boolean maxLifeReached(Slot slot) {
-		return maxLifeTime != -1 && slot.lifeTime() >= maxLifeTime;
+	boolean testEvictionPredicate(Slot slot) {
+		return poolConfig.evictionPredicate().test(slot.connection, new PooledSlotMetadata(slot));
 	}
```

Comment:
@samueldlightfoot , 

First of all, thanks for the hard work !

I have one comment: since I'm currently working on optimizing the Http2Pool class, which is a critical component, is it possible to avoid allocating the `PooledSlotMetadata` from the testEvictionPredicate method line 607 ? 

Instead of that, it seams feasible to manage to make the `Slot` inner class implementing itself the PooledRefMetadata interface, you could then avoid the allocation line 607 and replace it by:

```
		return poolConfig.evictionPredicate().test(slot.connection, slot);
```

WDYT ?

Reply:
Great suggestion - I have pushed the changes.